react-components/src/components/App/app.js

```diff
@@ -20,24 +20,33 @@ import ApolloClient from 'apollo-boost';
 import { CartProvider, CartInitializer } from '../Minicart';
 import { CheckoutProvider } from '../Checkout';
 import UserContextProvider from '../../context/UserContext';
+import { checkCookie, cookieValue } from '../../utils/cookieUtils';
 
 const App = props => {
     const { uri, storeView = 'default' } = props;
 
     const client = new ApolloClient({
         uri,
-        headers: { Store: storeView }
+        headers: { Store: storeView },
+        request: operation => {
+            let token = checkCookie('cif.userToken') ? cookieValue('cif.userToken') : '';
+            operation.setContext({
```

Comment:
Ideally we should only set the header if a token is set. Otherwise we end up with an invalid `'Authorization: Bearer '` header.

Reply:
Ack